manifest.toml

```
@@ -0,0 +1,115 @@
+default = true
+
+[install.encryption]
+ask.en = "Enable end-to-bridge encryption?"
+ask.fr = "Activer le chiffrement entre Synapse et le bridge ?"
+type = "boolean"
+help.en = "Only activate if you know the prerequisites and constraints related to E2B."
+help.fr = "N'activer que si vous connaissez les prérequis et constraintes liées à E2B."
+default = true
+
+[install.botadmin]
+ask.en = "Choose the Matrix account administrator of the WhatsApp bot"
+ask.fr = "Choisissez le compte Matrix administrateur du robot WhatsApp"
+type = "string"
+example = "@johndoe:localsynapse.servername or @johndoe:matrix.org"
+help.en = "The administrator does not need to be a local Synapse account. Valid formats are @johndoe:localsynapse.servername or @johndoe:matrix.org"
+help.fr = "L'administrateur peut ne pas être un compte local Synapse. Les formats valables sont @johndoe:localsynapse.servername or @johndoe:matrix.org"
+
+[install.botusers]
+ask.en = "Choose Matrix user(s) authorized to bridge with the WhatsApp bot."
+ask.fr = "Choisissez le/les compte(s) Matrix autorisés à utiliser la passerelle WhatsApp."
+type = "string"
+example = "@johndoe:server.name or server.name or *"
+help.en = """
+A remote or local user (@johndoe:server.name),the local server (server.name), a remote server (matrix.org), or all remote/local servers (*) can be authorized.
+Give the Matrix server_name, not the full domain/URL.
+It is also possible to specify multiple values by separating them with comma. Example: @johndoe:server.name,domain.tld,matrix.org"""
+help.fr = """Un compte local ou distant (@johndoe:server.name), le serveur local (server.name), un serveur distant (matrix.org), ou tous les serveurs remote/local (*).
+Donner le nom du serveur Matrix, pas le domaine/URL complet.
+Il est également possible de spécifier plusieurs valeurs en les séparant par une virgule. Exemple : @johndoe:server.name,domain.tld,matrix.org"""
+
+[resources]
+
+[resources.apt]
+packages = "g++, postgresql, ffmpeg"
+
+[resources.database]
+type = "postgresql"
+
+[resources.permissions]
+main.allowed = "all_users"
+main.auth_header = false
+
+[resources.system_user]
+home = "/opt/yunohost/__APP__"
+
+[resources.install_dir]
+dir = "/opt/yunohost/__APP__"
+
+[resources.ports]
+main.default = 8449
+
+[resources.sources]
+
+[resources.sources.main]
+in_subdir = false
+extract = false
+rename = "mautrix-whatsapp"
+amd64.url = "https://github.com/mautrix/whatsapp/releases/download/v0.9.0/mautrix-whatsapp-amd64"
+amd64.sha256 = "27b5d3ee0cada177207b662072b42923956e38a9c10c4e167eeca17881f3bffb"
+arm64.url = "https://github.com/mautrix/whatsapp/releases/download/v0.9.0/mautrix-whatsapp-arm64"
+arm64.sha256 = "dc475a2af45988d6d428af18c5f2a30da24930201e0e667a0b749a198a3bd02d"
+armhf.url = "https://github.com/mautrix/whatsapp/releases/download/v0.9.0/mautrix-whatsapp-arm"
+armhf.sha256 = "22c28d17ec868c4579e8ed64e092b0947a268c82ed9714d3ae45ace05b477fd8"
```

With packaging v2, we can drop the updater.sh script and use this:
```suggestion
armhf.sha256 = "22c28d17ec868c4579e8ed64e092b0947a268c82ed9714d3ae45ace05b477fd8"
autoupdate.strategy = "latest_github_release"
autoupdate.asset.amd64 = "mautrix-whatsapp-amd64"
autoupdate.asset.arm64 = "mautrix-whatsapp-arm64"
autoupdate.asset.armhf = "mautrix-whatsapp-arm"
```
See https://yunohost.org/fr/packaging_apps_resources#regarding-autoupdate

---

Thanks for the tip 🙂 but is it possible to have any custom step done with this approach? because in the current `updater.sh` script I have some custom steps to also autoupdate the `config.yaml` file which makes it much easier to make the update and I would like to keep it that way 😣